### ISSUE TITLE
OWSLib 0.26 issue with PyProj

### ISSUE THREAD

**Issue by cheginit** [geopython/OWSLib#823]
The new `owslib` version limits `pyproj` to < 3.3 which causes an issue with `rasterio` and `gdal` versions on macOS systems. Therefore, on macOS `owslib` cannot be used with `rasterio` and `gdal`. For example, you can check out the details of the log in one of my packages that depends on `owslib` and fails with `owslib=0.26` [here](https://github.com/cheginit/pygeohydro/runs/6872978939?check_suite_focus=true). I temporarily pinned `owslib` in my package to 0.25 to see how the issue with `pyproj` > 3.3 can be solved in `owslib=0.26`.

**Comment by SpacemanPaul** [comment]
Ubuntu 22.04 packages pyproj 3.3.0 as a deb too.

**Comment by kalxas** [comment]
Yes we need a new release soon to fix the issue

**Comment by pvgenuchten** [comment]
with the new release 0.27.2 the pyproj issue seem resolved

however i now get this failure when upgrading owslib

is it related or something on my environment with gdal

```
Updating rasterio (1.3b1 -> 1.2.10): Failed
```
```
        building 'rasterio._base' extension
        rasterio/_base.c:4675:26: warning: assigning to 'char *' from 'const char *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
            __pyx_v_c_drivername = __pyx_f_8rasterio_5_base_get_driver_name(__pyx_v_driver);
                                 ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        rasterio/_base.c:4750:26: warning: assigning to 'char *' from 'const char *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
            __pyx_v_c_extensions = GDALGetMetadataItem(__pyx_v_driver, ((char const *)"DMD_EXTENSIONS"), NULL);
```

**Comment by pvgenuchten** [comment]
issue is resolved for me in 0.27.2, thanx @kalxas

**Comment by cheginit** [comment]
I also confirm that the issue in 0.27.2 both on pypi and conda-forge are resolved now.
